asset-transfer-basic/chaincode-javascript/test/assetTransfer.test.js

```diff
@@ -0,0 +1,251 @@
+        });
+    });
+
+    afterEach(() => {
+    });
+
+    describe('Test InitLedger', () => {
+        it('should return error on InitLedger', async () => {
+            chaincodeStub.putState.rejects('failed inserting key');
+            let assetTransfer = new AssetTransfer();
+            try {
+                await assetTransfer.InitLedger(transactionContext);
+                assert.fail('InitLedger should have failed');
+            } catch (err) {
+                expect(err.name).to.equal('failed inserting key');
+            }
+        });
+
+        it('should return success on InitLedger', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.InitLedger(transactionContext);
+            let ret = JSON.parse((await chaincodeStub.getState('asset6')).toString());
+            expect(ret.ID).to.equal('asset6');
+        });
+    });
+
+    describe('Test CreateAsset', () => {
+        it('should return error on CreateAsset', async () => {
+            chaincodeStub.putState.rejects('failed inserting key');
+
+            let assetTransfer = new AssetTransfer();
+            try {
+                await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+                assert.fail('CreateAsset should have failed');
+            } catch(err) {
+                expect(err.name).to.equal('failed inserting key');
+            }
+        });
+
+        it('should return success on CreateAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            let ret = JSON.parse((await chaincodeStub.getState('asset1')).toString());
+            expect(ret.ID).to.equal('asset1');
+            expect(ret.Color).to.equal('blue');
+            expect(ret.Size).to.equal(5);
+            expect(ret.Owner).to.equal('Tomoko');
+            expect(ret.AppraisedValue).to.equal(300);
+        });
+    });
+
+    describe('Test ReadAsset', () => {
+        it('should return error on ReadAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            try {
+                await assetTransfer.ReadAsset(transactionContext, 'asset2');
+                assert.fail('ReadAsset should have failed');
+            } catch (err) {
+                expect(err.message).to.equal('The asset asset2 does not exist');
+            }
+        });
+
+        it('should return success on ReadAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            let ret = JSON.parse(await chaincodeStub.getState('asset1'));
+            expect(ret.ID).to.equal('asset1');
```

Comment:
Compare the ret with the expected returned object using `.eql` as noted below

Reply:
Fixed as indicated